src/helpers/themeStyle.ts

```diff
@@ -1,16 +1,26 @@
 export function themeStyle(currentTheme, tagName, styleThis, el) {
-  const css = currentTheme ? currentTheme.components[tagName] : '';
+  /*
+    Helper function that will appened a stylesheet with scoped styling for specific component
+    either in the shadowRoot with adoptedStyleSheet API
+    or with a fallback method (IE,Old edge and so on)
+  */
+
   let style;
+  const css = currentTheme ? currentTheme.components[tagName] : '';
 
-  if(!styleThis) return;
+  // Fallback for currentTheme, initially empty or the currentTheme doesn't contain a version property
+  if(!currentTheme || !currentTheme.version || !styleThis) {
+    return;
+  }
 
-  // This is used by browsers with support for shadowdom
+  // If the browser has support for adoptedStyleSheet (Chromium)
```

Comment:
The previous statement is actually correct. Because el.shadowRoot is only exist for browser that support shadow DOM. And adaptedStyleSheet is part of that.

Reply:
The condition check if the browser has support specifically for adoptedStyletSheet not shadowDOM. 

ShadowDOM is supported by a lot browsers but not a lot of browsers support adoptedStyleSheet API.